File: Week 1/caesar.cpp
#include<iostream>
#include<string>
#include <cctype>
using namespace std;

void encryption(string plaintext,int key){
    string ciphertext ="";
    for (char c : plaintext){
        if(isupper(c)){
            ciphertext +=char((c-'A'+key)%26 + 'A');
        }
        else if(islower(c)){
            ciphertext +=char((c-'a'+key)%26 + 'a');
        }
        else{
            ciphertext += c;
        }        
    }
    cout<<ciphertext;
}

int main(){
    std::string text;
    int key;
    char choice;
    cout << "Enter text: ";
    getline(cin, text);
    cout << "Enter shift key: ";
    cin >> key;
    cout << "What do you want to do? E/D :";
    cin>>choice;
    if(choice == 'E'){
        encryption(text,key);
    }
    else if(choice == 'D'){
        encryption(text,-key);
    }
    else{
        cout<<"Enter Valid Choice !";
    }
    
   return 0; 
}

File: Week 1/caesar.py
def encryption(text,key):
    output =""
    result = ""
    for c in text:
        if c.isupper():
            result += chr((ord(c) - 65 + key) % 26 + 65)
        elif c.islower():
            result += chr((ord(c) - 97 + key) % 26 + 97)
        else:
            result += c
    print(result)

text = input("Enter text: ")
key = int(input("Enter Key: "))
choice = input("What do you want to do? E/D :")

if choice=='E':
    encryption(text,key)
elif choice=='D':
    encryption(text,-key)
else:
    print("Enter Valid Choice !")
